tests/lib/rules/no-unknown-property.js

```diff
@@ -573,6 +577,18 @@ ruleTester.run('no-unknown-property', rule, {
         },
       ],
     },
+    {
+      code: '<div data-UPPERCASE="invalid" />',
+      errors: [
+        {
+          messageId: 'unknownProp',
```

Comment:
i think this case should have its own error message, explaining the same thing react’s does - it’s not exactly unknown, it’s valid in html.